### ISSUE TITLE
Displaying wrong slides when using minSlides /maxSlides and Loop

### ISSUE THREAD

**Issue by Kell** [stevenwanderski/bxslider-4#953]
Hey,

just tried out this slider with some easy examples and realized that it shows me wrong Slides when Slider displaying 1 slide.

$('.bxslider-small').bxSlider({
	slideWidth: 500,
	minSlides: 1,
	maxSlides: 4,
	slideMargin: 10
});

i have an ul with 4 images. it all seems fine until i  resize the window so the slider shows me 1 Slide.
When i use the "next" arrow i see image 1, image 3, image 2, image 1 then it starts again with Image 1 with the same order.
This seems only to happen when infiniteLoop is true. if set  to false i get the right image order.

**Comment by mattcph** [comment]
seems the object indexing for infiniteLoop is broken or something, have not looked but confirm this issue is reproducable.

**Comment by enoversum** [comment]
I guess it is not the quite the same, but I have the issue that the slider skips visibly one of the slider items on the 'full' mode (i.e. desktop with max slides showing). I have 7 elements, have maxSlides set to 5, minSlides to 3, infiniteLoop to default (= true). It rightly shows slides 1, 2, 3, 4 and 5. When I click on the pager, it jumps over slide 6 and shows item 7 and then item 1, 2, 3, 4. Once I disable the infiniteLoop, it is working again. Reproducable.

**Comment by mattcph** [comment]
Same issue, I expect the build logic for infinite slide is having issues. Still have not had time to look at it, if you can pin point the slide build location perhaps we can patch it.

**Comment by tomchiverton** [comment]
Yup. infinateLoop is broken when the final slide does not contain the same number of items as the previous slide. Avoid.
